fenwick/src/intTest/java/org/opencadc/fenwick/StorageSiteSyncTest.java

```diff
@@ -0,0 +1,373 @@
+
+import javax.security.auth.Subject;
+
+
+public class StorageSiteSyncTest {
+    private static final File PROXY_PEM = new File(System.getProperty("user.home") + "/.ssl/cadcproxy.pem");
+    private static final Logger LOGGER = Logger.getLogger(StorageSiteSyncTest.class);
+
+    static {
+        Log4jInit.setLevel("org.opencadc.inventory", Level.INFO);
+        Log4jInit.setLevel("ca.nrc.cadc.db", Level.INFO);
+        Log4jInit.setLevel("org.opencadc.fenwick", Level.DEBUG);
+    }
+
+    private final StorageSiteDAO storageSiteDAO = new StorageSiteDAO();
+    private final ArtifactDAO artifactDAO = new ArtifactDAO();
+
+    public StorageSiteSyncTest() throws Exception {
+        final DBConfig dbConfig = new DBConfig();
+        final ConnectionConfig cc = dbConfig.getConnectionConfig(TestUtil.SERVER, TestUtil.DATABASE);
+        DBUtil.createJNDIDataSource("jdbc/StorageSiteSyncTest", cc);
+
+        final Map<String,Object> config = new TreeMap<>();
+        config.put(SQLGenerator.class.getName(), SQLGenerator.class);
+        config.put("jndiDataSourceName", "jdbc/StorageSiteSyncTest");
+        config.put("database", TestUtil.DATABASE);
+        config.put("schema", TestUtil.SCHEMA);
+
+        storageSiteDAO.setConfig(config);
+        artifactDAO.setConfig(config);
+    }
+
+    private void wipe_clean(ResourceIterator<Artifact> artifactIterator) throws Exception {
+        while (artifactIterator.hasNext()) {
+            Artifact a = artifactIterator.next();
+            LOGGER.debug("deleting test uri: " + a.getURI() + " ID: " + a.getID());
+            artifactDAO.delete(a.getID());
+        }
+    }
+
+    @Before
+    public void cleanTestEnvironment() throws Exception {
+        LOGGER.debug("cleaning stored artifacts...");
+        ResourceIterator<Artifact> storedArtifacts = artifactDAO.storedIterator(null);
+        wipe_clean(storedArtifacts);
+        storedArtifacts.close();
+
+        LOGGER.debug("cleaning unstored artifacts...");
+        ResourceIterator<Artifact> unstoredArtifacts = artifactDAO.unstoredIterator(null);
+        wipe_clean(unstoredArtifacts);
+        unstoredArtifacts.close();
+
+        LOGGER.debug("Wiping Storage Sites.");
+        for (final StorageSite storageSite : storageSiteDAO.list()) {
+            LOGGER.debug(String.format("Removing storage site '%s'.", storageSite.getName()));
+            storageSiteDAO.delete(storageSite.getID());
+        }
+    }
+
+    /**
+     * Test StorageSite synchronization.
+     * Test one: Populate the local inventory database with a StorageSite from the remote Luskan.
+     * Test two: Ensure the same site is put again without error.
+     * Test three: Ensure a failed execution if that same site has a different checksum than its computed one.
+     * Test four: Ensure a failed execution if multiple sites are found.
+     * Test five: Ensure a failed execution if no sites are found.
+     *
+     * @throws Exception
+     */
+    @Test
```

Comment:
This needs to be broken up into separate test methods so it is easier to run some of the tests (eg when diagnosing an issue) and so the cursory summary printed by "gradle intTest" tells you what worked and what didn't -- if everything fails it is usually config.. A single method can do multiple things (like one and two since they build on each other) ... but each kind of fail should be it's own test.